diff --git a/drivers/gpu/arm/mali400/mali/linux/mali_linux_trace.h b/drivers/gpu/arm/mali400/mali/linux/mali_linux_trace.h
index e6c928dc..22226082 100755
--- a/drivers/gpu/arm/mali400/mali/linux/mali_linux_trace.h
+++ b/drivers/gpu/arm/mali400/mali/linux/mali_linux_trace.h
@@ -18,7 +18,6 @@
 
 #undef  TRACE_SYSTEM
 #define TRACE_SYSTEM mali
-#define TRACE_SYSTEM_STRING __stringfy(TRACE_SYSTEM)
 
 #define TRACE_INCLUDE_PATH .
 #define TRACE_INCLUDE_FILE mali_linux_trace
diff --git a/include/trace/trace_events.h b/include/trace/trace_events.h
index de996cf6..7d7608c8 100644
--- a/include/trace/trace_events.h
+++ b/include/trace/trace_events.h
@@ -24,7 +24,7 @@
 
 #define __app__(x, y) str__##x##y
 #define __app(x, y) __app__(x, y)
-
+#undef TRACE_SYSTEM_STRING
 #define TRACE_SYSTEM_STRING __app(TRACE_SYSTEM_VAR,__trace_system_name)
 
 #define TRACE_MAKE_SYSTEM_STR()				\
