diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index 96f021ca..9f3db17f 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -20,7 +20,7 @@
  * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
  * DEALINGS IN THE SOFTWARE.
  */
-
+#define DEBUG 1
 #include <linux/backlight.h>
 #include <linux/gpio/consumer.h>
 #include <linux/module.h>
@@ -800,7 +800,7 @@ static int panel_simple_probe(struct device *dev, const struct panel_desc *desc)
 	const char *cmd_type;
 	u32 val;
 	int err;
-
+  dev_info(dev, "init panel\n");
 	panel = devm_kzalloc(dev, sizeof(*panel), GFP_KERNEL);
 	if (!panel)
 		return -ENOMEM;
diff --git a/drivers/gpu/drm/rockchip/analogix_dp-rockchip.c b/drivers/gpu/drm/rockchip/analogix_dp-rockchip.c
index a8312cf0..9fd7ac81 100644
--- a/drivers/gpu/drm/rockchip/analogix_dp-rockchip.c
+++ b/drivers/gpu/drm/rockchip/analogix_dp-rockchip.c
@@ -367,6 +367,10 @@ static int rockchip_dp_bind(struct device *dev, struct device *master,
 			return -EINVAL;
 		}
 
+    dev_info(dev, "panel name:%s,type:%s, parent:%s.\n", 
+      panel_node->name, panel_node->type,
+      panel_node->parent->name); 
+
 		panel = of_drm_find_panel(panel_node);
 		if (!panel) {
 			DRM_ERROR("failed to find panel\n");
