diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/Kconfig b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/Kconfig
index 10c06951..b20738a1 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/Kconfig
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/Kconfig
@@ -8,14 +8,14 @@ config BCMDHD
 config BCMDHD_FW_PATH
 	depends on BCMDHD
 	string "Firmware path"
-	default "/system/etc/firmware/fw_bcmdhd.bin"
+	default "/lib/firmware/fw_bcmdhd.bin"
 	---help---
 	  Path to the firmware file.
 
 config BCMDHD_NVRAM_PATH
 	depends on BCMDHD
 	string "NVRAM path"
-	default "/system/etc/firmware/nvram.txt"
+	default "/lib/firmware/nvram.txt"
 	---help---
 	  Path to the calibration file.
 
diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/rk_wifi_config.c b/drivers/net/wireless/rockchip_wlan/rkwifi/rk_wifi_config.c
index fa07a94c..c39f437f 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/rk_wifi_config.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/rk_wifi_config.c
@@ -17,7 +17,7 @@
  * Set Firmware Path
  */
  
-#define VENDOR_ETC_FIRMWARE "/vendor/etc/firmware/"
+#define VENDOR_ETC_FIRMWARE "/lib/firmware/"
 #define SYSTEM_ETC_FIRMWARE "/system/etc/firmware/"
 char ANDROID_FW_PATH[64] = {0};
 
diff --git a/drivers/net/wireless/rockchip_wlan/ssv6xxx/firmware/ssv6051-wifi.cfg b/drivers/net/wireless/rockchip_wlan/ssv6xxx/firmware/ssv6051-wifi.cfg
index c072960f..f8fd984e 100644
--- a/drivers/net/wireless/rockchip_wlan/ssv6xxx/firmware/ssv6051-wifi.cfg
+++ b/drivers/net/wireless/rockchip_wlan/ssv6xxx/firmware/ssv6051-wifi.cfg
@@ -9,7 +9,7 @@
 # Priority.2 firmware_path
 # Priority.3 default firmware
 ##################################################
-firmware_path = /vendor/etc/firmware/
+firmware_path = /lib/firmware/
 
 ############################################################
 # MAC address
diff --git a/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/a33-wifi.cfg b/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/a33-wifi.cfg
index a4067cfb..d1237fc3 100644
--- a/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/a33-wifi.cfg
+++ b/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/a33-wifi.cfg
@@ -9,7 +9,7 @@
 # Priority.2 firmware_path
 # Priority.3 default firmware
 ##################################################
-firmware_path = /etc/firmware/
+firmware_path = /lib/firmware/
 
 ############################################################
 # MAC address
diff --git a/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/atm7039-actions-wifi.cfg b/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/atm7039-actions-wifi.cfg
index e3056691..e413b3dc 100644
--- a/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/atm7039-actions-wifi.cfg
+++ b/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/atm7039-actions-wifi.cfg
@@ -9,7 +9,7 @@
 # Priority.2 firmware_path
 # Priority.3 default firmware
 ##################################################
-firmware_path = /etc/firmware/
+firmware_path = /lib/firmware/
 
 ############################################################
 # MAC address
diff --git a/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/h3-wifi.cfg b/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/h3-wifi.cfg
index 9daefbf5..caf50eda 100644
--- a/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/h3-wifi.cfg
+++ b/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/h3-wifi.cfg
@@ -9,7 +9,7 @@
 # Priority.2 firmware_path
 # Priority.3 default firmware
 ##################################################
-firmware_path = /etc/firmware/
+firmware_path = /lib/firmware/
 
 ############################################################
 # MAC address
diff --git a/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/h8-wifi.cfg b/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/h8-wifi.cfg
index e65de637..a24e0b81 100644
--- a/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/h8-wifi.cfg
+++ b/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/h8-wifi.cfg
@@ -9,7 +9,7 @@
 # Priority.2 firmware_path
 # Priority.3 default firmware
 ##################################################
-firmware_path = /etc/firmware/
+firmware_path = /lib/firmware/
 
 ############################################################
 # MAC address
diff --git a/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/v66-wifi.cfg b/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/v66-wifi.cfg
index 79f83d18..afb3c7a0 100644
--- a/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/v66-wifi.cfg
+++ b/drivers/net/wireless/rockchip_wlan/ssv6xxx/platforms/v66-wifi.cfg
@@ -9,7 +9,7 @@
 # Priority.2 firmware_path
 # Priority.3 default firmware
 ##################################################
-firmware_path = /etc/firmware/
+firmware_path = /lib/firmware/
 
 ############################################################
 # MAC address
diff --git a/drivers/net/wireless/rockchip_wlan/ssv6xxx/ssvdevice/ssvdevice.c b/drivers/net/wireless/rockchip_wlan/ssv6xxx/ssvdevice/ssvdevice.c
index d94a5f50..afce4ece 100644
--- a/drivers/net/wireless/rockchip_wlan/ssv6xxx/ssvdevice/ssvdevice.c
+++ b/drivers/net/wireless/rockchip_wlan/ssv6xxx/ssvdevice/ssvdevice.c
@@ -58,7 +58,7 @@ static char *ssv6xxx_cmd_buf;
 char *ssv6xxx_result_buf;
 extern struct ssv6xxx_cfg_cmd_table cfg_cmds[];
 extern struct ssv6xxx_cfg ssv_cfg;
-char DEFAULT_CFG_PATH[] = "/vendor/etc/firmware/ssv6051-wifi.cfg";
+char DEFAULT_CFG_PATH[] = "/lib/firmware/ssv6051-wifi.cfg";
 static int ssv6xxx_dbg_open(struct inode *inode, struct file *filp)
 {
     filp->private_data = inode->i_private;
