diff --git a/vm_tools/init/vm_concierge.conf b/vm_tools/init/vm_concierge.conf
index 3a6cb856b6..c5796818b3 100644
--- a/vm_tools/init/vm_concierge.conf
+++ b/vm_tools/init/vm_concierge.conf
@@ -94,7 +94,7 @@ pre-start script
 
   if crossystem "cros_debug?1"; then
     touch /run/vm/dev_mode
-    if [! -d "${ARCVM_ADB_SYNC_DIR}"]; then
+    if [ ! -d "${ARCVM_ADB_SYNC_DIR}" ]; then
       mkdir -p "${ARCVM_ADB_SYNC_DIR}"
       chown crosvm:crosvm "${ARCVM_ADB_SYNC_DIR}"
     fi
