diff --git a/concierge/vm_util.cc b/concierge/vm_util.cc
index 8b3376f343..d61f6fe8e6 100644
--- a/concierge/vm_util.cc
+++ b/concierge/vm_util.cc
@@ -179,9 +179,9 @@ std::string GetVmMemoryMiB() {
   int64_t sys_memory_mb = base::SysInfo::AmountOfPhysicalMemoryMB();
   int64_t vm_memory_mb;
   if (sys_memory_mb >= 4096) {
-    vm_memory_mb = sys_memory_mb - 1024;
+    vm_memory_mb = sys_memory_mb - 2048;
   } else {
-    vm_memory_mb = sys_memory_mb / 4 * 3;
+    vm_memory_mb = sys_memory_mb / 3 * 2;
   }
 
   // Limit guest memory size to avoid running out of host process address space.
